Solitaire/js/main.js

```diff
@@ -1,4 +1,259 @@
-
+(function () {
+    var $btnReset = document.getElementById('reset');
+    var cards = [];
+	var playingDecks = {
+		deckPlaying1: {array: [], cardPos: 1, click: checkCard},
+		deckPlaying2: {array: [], cardPos: 3, click: checkCard},
+		deckPlaying3: {array: [], cardPos: 6, click: checkCard},
+		deckPlaying4: {array: [], cardPos: 10, click: checkCard},
+		deckPlaying5: {array: [], cardPos: 15, click: checkCard},
+		deckPlaying6: {array: [], cardPos: 21, click: checkCard},
+		deckPlaying7: {array: [], cardPos: 28, click: checkCard},
+		deckDistribution: {array: [], cardPos: 52, click: openCardDestribution},
+		deckDistributionOpen: {array: [], click: chooseCardDistribution},
+		deckFinish1: {array: [], click: checkCard},
+		deckFinish2: {array: [], click: checkCard},
+		deckFinish3: {array: [], click: checkCard},
+		deckFinish4: {array: [], click: checkCard}
+	}
+	var mixRandom = function() {
+        return Math.random() - 0.5;
+    };
+	var deckSelected = null;
+    var cardSelectedNumber = null;
+
+    
+    function fillingArrayCards() {
+
+        for (var i = 1; i < 53; i++) {
+            var value = i%13 === 0 ? 13 : i%13;
+            var name;
+            var suit;
+            var suitGroup = Math.ceil(i/13);
+    
+            if (suitGroup === 1) {
+                suit = 'hearts';
+            } else if (suitGroup === 2) {
+                suit = 'diamonds';
+            } else if (suitGroup === 3) {
+                suit = 'clovers';
+            } else if (suitGroup === 4) {
+                suit = 'spades';
+            }
+    
+            if (value === 1) {
+                name = 'A';
+            } else if (value === 11) {
+                name = 'J';
+            } else if (value === 12) {
+                name = 'Q';
+            } else if (value === 13) {
+                name = 'K';
+            } else {
+                name = String(value);
+            }
+    
+            cards.push({
+                isVisible: false,
+                id: 'card' + i,
+                value: value,
+                name: name,
+                suit: suit,
+            });
+        }
+    };
+
+    function fillingGameDecks() {
+		for (var i = 0; i < cards.length; i++) {
```

Comment:
The function has 2 nested loops with conditions. It would be good to refactor this.